build/version.go

```diff
@@ -101,3 +105,24 @@ func Tags() []string {
 
 	return strings.Split(RawTags, ",")
 }
+
+// WithBuildInfo derives a child context with the build information attached as
+// attributes. At the moment, this only includes the current build's commit
+// hash.
+func WithBuildInfo(ctx context.Context, cfg *LogConfig) (context.Context,
+	error) {
+
+	if cfg.NoCommitHash {
+		return ctx, nil
+	}
+
+	// Convert the commit hash to a byte slice.
+	commitHash, err := hex.DecodeString(CommitHash)
+	if err != nil {
+		return nil, fmt.Errorf("unable to decode commit hash: %w", err)
+	}
+
+	// Include the first 6 bytes of the commit hash in the context as an
+	// slog attribute.
+	return btclog.WithCtx(ctx, btclog.Hex6("commit_hash", commitHash)), nil
```

Comment:
I think a `Hex2` or `Hex3` would be sufficient here. 6 bytes means 12 characters on potentially each line. And perhaps we can shorten the `commit_hash` as well? Doesn't need to be that obvious, then it's perhaps also less likely for people to complain about it. How about just `rev=aabbcc`?

Reply:
cool yeah good idea - updated